src/analysis/retail/monk/mistweaver/modules/spells/Revival.tsx

```diff
@@ -180,18 +184,36 @@ class Revival extends Analyzer {
           });
           const allPerfs = [efPerf];
           if (this.selectedCombatant.hasTalent(TALENTS_MONK.SHAOHAOS_LESSONS_TALENT)) {
-            let lessonPerf = QualitativePerformance.Good;
-            if (!cast.lessonsBuffActive) {
-              lessonPerf = QualitativePerformance.Fail;
+            let lessonPerf = QualitativePerformance.Fail;
+            if (
+              cast.lessonsBuffActive ||
+              this.shaohaos.getNextBuff() !== SPELLS.LESSON_OF_DOUBT_BUFF
```

Comment:
I would prefer to change this to any buff except Fear since its the only one that doesn't provide substantial value to revival